### ISSUE TITLE
mriqc docker Cannot allocate memory (v 0.9.4)

### ISSUE THREAD

**Issue by jnecus** [poldracklab/mriqc#536]
Hello,

I frequently get these errors when trying to run mriqc using docker:

```
Standard error: Image Exception : #99 :: Out of memory
Cannot allocate memory
```

I am trying to run over 100 anatomical T1w subjects using a machine with 8 cpu's and 64gb RAM.

I wondered whether there is a way to limit the number of images being processed in parallel when using docker and what this command would look like?

Thank you for your help,
Joe

**Comment by oesteban** [comment]
Hi @jnecus,

You can set a limitation on memory used with the option [``--mem_gb 64``](http://mriqc.readthedocs.io/en/stable/running.html).

You can reduce the number of subjects processed in parallel using the option [``--n_procs``](http://mriqc.readthedocs.io/en/stable/running.html). Please make sure that the option ``--ants-nthreads`` is less or equal to ``--n_procs``, except for ``--n_procs`` == 1 in which case you can use your 8 cpus (``--ants-nthreads 8``) for registration.

You can pin the execution to a limited amount of subjects with the argument ``--participant_label 001 002``.

If you are using Docker on MacOSX, you will likely need to [raise the memory available to docker containers](http://mriqc.readthedocs.io/en/stable/docker.html#preliminaries).

Please let us know if this is still failing.

**Comment by chrisfilo** [comment]
Another thing worth looking into is any other processes (potentially run by other users) running on the machine and consuming memory.

**Comment by jnecus** [comment]
Thank you both. Will let you know how I get on.



________________________________

**Comment by jnecus** [comment]
Just an update,

It failed once again and then after re-running a third time seemed to work but I'm not sure why. 

In each case when outputting derivatives and reports at the end all the RAM on the computer was used, rendering the computer temporarily unusable. This occurred despite using the --mem_gb flag.Here is an example of the command that I used:

`sudo docker run -it --rm -v /my/data/path/:/data:ro -v /my/output/path:/out poldracklab/mriqc:0.9.4 /data /out participant --participant_label $subs --n_procs 6 --ants-nthreads 6 --mem_gb 56 
`

**Comment by oesteban** [comment]
Hi @jnecus, sorry for coming back to this late. Have you tried numbers below 56GB? As Chris pointed out, some other processes could be taking up some memory and drive the SO to start swapping.

**Comment by spanta28** [comment]
You can go to docker preferences, and increase the memory available to the docker and see if that works. It worked for me.

**Comment by oesteban** [comment]
Hi @jnecus, did this problem persist?

could you also report the output of this command?

```
sudo docker run -it --rm -v /my/data/path/:/data:ro -v /my/output/path:/out --entrypoint=/bin/bash poldracklab/mriqc:0.9.4 cat /proc/sys/vm/overcommit_memory
```

**Comment by jnecus** [comment]
Sorry, I stopped using this a while back due to the above issue.

**Comment by chrisfilo** [comment]
Sorry it did not work out - we are actively working on resolving the memory issues.

**Comment by jnecus** [comment]
Ah, sorry for closing the issue.  I will re-post on here if I get around to testing it.

**Comment by chrisfilo** [comment]
Thank you - we appreciate your feedback.

Best,
Chris

**Comment by oesteban** [comment]
This issue was closed in #736 and reopened when I removed the stale bot. Apologies for the noise.
